src/lib.rs

```diff
@@ -124,89 +169,94 @@ pub fn get_local_timestamp_rfc3339() -> Result<String, Error> {
 /// * utc_offset - A caller specified offset
 ///
 /// # Returns
-/// Returns a `Result` of either the timestamp in the following format
-/// "[year]-[month]-[day]T[hour]:[minute]:[second][offset_hour
-/// sign:mandatory]:[offset_second]", or an error if the method fails.
+/// Returns a `Result` timestamp in the following format or the error encountered during its construction.
+/// ```text
+/// [year]-[month]-[day]T[hour]:[minute]:[second][offset_hour sign:mandatory]:[offset_second]
+/// ```
 #[allow(clippy::cast_lossless)]
-pub fn get_local_timestamp_from_offset_rfc3339(utc_offset: UtcOffset) -> Result<String, Error> {
-    let datetime_now = OffsetDateTime::now_utc();
-    if utc_offset != UtcOffset::UTC {
-        if let Some(t) = datetime_now.checked_add(Duration::hours(utc_offset.whole_hours() as i64))
-        {
-            let offset_datetime_now = t.replace_offset(utc_offset);
-            return Ok(offset_datetime_now.format(&TIME_FORMAT)?);
+#[inline]
+pub fn get_local_timestamp_from_offset_rfc3339(utc_offset: UtcOffset) -> Result<String> {
+    let dt_now = OffsetDateTime::now_utc();
+    let offset_dt_now = if utc_offset == UtcOffset::UTC {
+        dt_now
+    } else {
+        // verify: I changed this to minutes resolution, any reason it used to be `whole_minutes`?
```

Comment:
@gadunga

Reply:
meant to say whole_hours. Is this ok?

Reply:
I think I went with `whole_hours` originally was because I think the most common offsets are in whole hours, so I didn't think the resolution was necessary. Going with minutes is probably fine, especially if there are offsets which are not whole hours.